lib/matplotlib/axes/_axes.py

```diff
@@ -4013,7 +4013,9 @@ def dopatch(xs, ys, **kwargs):
         return dict(whiskers=whiskers, caps=caps, boxes=boxes,
                     medians=medians, fliers=fliers, means=means)
 
-    def _parse_scatter_color_args(self, c, edgecolors, kwargs, xshape, yshape):
+    @staticmethod
+    def _parse_scatter_color_args(c, edgecolors, kwargs, xshape, yshape,
+                                  get_next_color_func):
```

Comment:
Couldn't this be an optional kwarg to save the mysterious kwarg in the tests?  

If the kwarg stays in the tests, can you comment that its a dummy function...

We had someone be pretty firm that the tests should be of user-facing features, and not testing internal methods.  I'm not sure I 100% buy that idea, but it is something to consider here.  Does testing `ax.scatter` directly tests that no-one messes up the parsing before or after this fcn?

Reply:
> Couldn't this be an optional kwarg to save the mysterious kwarg in the tests?

The only usage of this function is to extract the parameter parsing, and in that call the argument is mandatory. I don't want to modify the signature to make the impression that a general use is possible without this parameter. The tests don't use it because we control inputs and thus the code path, but that's rather a consequence of the things we want to test and shouldn't affect the signature.

> We had someone be pretty firm that the tests should be of user-facing features, and not testing internal methods. I'm not sure I 100% buy that idea, but it is something to consider here. Does testing `ax.scatter` directly tests that no-one messes up the parsing before or after this fcn?

Generally I agree that you should only test the public interface. However, this is really expensive here, because you would have to generate a plot for each parameter. Since the color parameters are quite involved, they need a lot of testing. So we're splitting the function into two parts: 1. parse the color arguments to facecolors and edgecolors, 2. draw with these colors.

Part 1 can be tested separately and efficiently. It's still implicitly part of the public API. If something goes wrong in here, the plot will be broken. Part 2 needs testing as well to make sure no-one messes up the parsing before or after part 1. But that can happen with just a few simple tests. Also this should test the actual plotting.

The modified tests were only a subset of part 1 anyway. They did only smoke-test if the parameters were accepted or not.